.github/workflows/release-prep.yaml

```diff
@@ -0,0 +1,71 @@
+name: Prep Heat Release
+
+on:
+  issues:
+    types: [opened]
+env:
+  working_branch: workflows/release-prep
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7 # v2.10.1
+        with:
+          egress-policy: audit
+
+      - uses: actions/checkout@v3
+      - name: Extract Version Information
+        id: extract-version
+        if: contains(github.event.issue.labels.*.name, 'release-prep')
+        run: |
+          issue_title="${{ github.event.issue.title }}"
+          major=$(echo "$issue_title" | cut -d '.' -f 1)
+          minor=$(echo "$issue_title" | cut -d '.' -f 2)
+          micro=$(echo "$issue_title" | cut -d '.' -f 3)
+          echo "major=$major" >> $GITHUB_OUTPUT
+          echo "minor=$minor" >> $GITHUB_OUTPUT
+          echo "micro=$micro" >> $GITHUB_OUTPUT
+      - name: Delete Existing Version Update Branch
+        run: git push --delete origin ${{ env.working_branch }} || true
+      - name: Create Release Branch
+        if: steps.extract-version.outputs.micro == 0
+        run: |
+          git checkout -b release/${{ steps.extract-version.outputs.major }}.${{ steps.extract-version.outputs.minor }}.x
```

Comment:
I would add an extra `git checkout main; git pull` to make sure we are branching of the most recent version of main.